tools/contrib/fsurdat_modifier.py

```diff
@@ -0,0 +1,356 @@
+                    default=7)
+        parser.add_argument('--uni_snow', 
+                    help='Turn on the flag to create uniform snowpack. [default: %(default)s]', 
+                    action="store", 
+                    dest="uni_snow",
+                    default=False)
+        parser.add_argument('--overwrite_single_pft', 
+                    help='Turn on the flag to make the whole grid 100%% single PFT. [default: %(default)s]', 
+                    action="store", 
+                    dest="overwrite_single_pft",
+                    default=False)
+        parser.add_argument('--zero_nonveg', 
+                    help='Set all non-vegetation landunits to zero. [default: %(default)s]', 
+                    action="store", 
+                    dest="zero_nonveg",
+                    type =bool,
+                    default=False)
+        parser.add_argument('--no_saturation_excess', 
+                    help='Turn off saturation excess. [default: %(default)s]', 
+                    action="store", 
+                    dest="no_saturation_excess",
+                    type =bool,
+                    default=False)
+        parser.add_argument('--fsurdat_in', 
+                    help = 'Input surface dataset. [default: %(default)s]', 
+                    action = "store", 
+                    dest = "fsurdat_in",
+                    type = str,
+                    default = "/glade/p/cesmdata/cseg/inputdata/lnd/clm2/surfdata_map/release-clm5.0.18/surfdata_0.9x1.25_hist_78pfts_CMIP6_simyr2000_c190214.nc")
+        parser.add_argument('--fsurdat_out', 
+                    help = 'Output surface dataset. [default: %(default)s]', 
+                    action = "store", 
+                    dest = "fsurdat_out",
+                    type = str,
+                    default = "/glade/scratch/" + myname + "/surfdata_0.9x1.25_hist_78pfts_CMIP6_simyr2000_c190214_modified.nc")
+
+        return parser
+
+
+def get_git_sha():
+    """
+    Returns Git short SHA for the currect directory.
+    """
+    return subprocess.check_output(['git', 'rev-parse', '--short', 'HEAD']).strip().decode() 
+
+
+class ModifyFsurdat:
+    """
+    A case to encapsulate function that modifies fsurdat file.
+
+    ...
+
+    Attributes
+    ----------
+
+    Methods
+    -------
+    add_tag_to_filename(filename, tag)
+       add a tag and timetag to a filename ending with 
+       [._]cYYMMDD.nc or [._]YYMMDD.nc 
+
+    update_metadata(nc, filename)
+
+    modify:
+        Modify input surface dataset
+    """
+
+    def __init__(self, overwrite_single_pft, dominant_pft,
+                 zero_nonveg_landunits, uniform_snowpack, no_saturation_excess):
+        super().__init__()
+        self.overwrite_single_pft = overwrite_single_pft
+        self.dominant_pft = dominant_pft
+        self.zero_nonveg_landunits = zero_nonveg_landunits
+        self.uniform_snowpack = uniform_snowpack
+        self.no_saturation_excess = no_saturation_excess
```

Comment:
- [ ] The member variables of the class should probably be "private" (leading underscore) to note that they shouldn't be accessed directly from outside the class. e.g., `self._overwrite_single_pft`.